packages/react-storage/src/components/StorageManager/hooks/useUploadFiles/useUploadFiles.ts

```diff
@@ -71,22 +71,25 @@ export function useUploadFiles({
       };
 
       if (file) {
-        resolveFile({ processFile, file, key }).then(({ key, file }) => {
-          onUploadStart?.({ key });
-          const uploadTask = uploadFile({
-            file,
-            key,
-            level: accessLevel,
-            progressCallback: onProgress,
-            errorCallback: onError,
-            completeCallback: onComplete,
-          });
+        resolveFile({ processFile, file, key }).then(
+          ({ key: processedKey, file, ...rest }) => {
```

Comment:
Was there an issue leaving `file` in `rest` and spreading?

Reply:
ugh no, but I just committed your suggestion which had `file` destructured out. Removed `file` so now its part of `rest`